### PR TITLE
[edward] Fixed wsy/wsz loss multipliers — direct binding-constraint attack

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

**Fixed per-channel loss multipliers (`wsy×3, wsz×5, others×1`) directly attack the wsy/wsz binding constraint without the optimization instability of GradNorm or the complexity of adaptive reweighting.**

The wsy/wsz axes are the dominant gap in the Wave 1 winner:
- wsy = 9.10% (target 3.65, gap = 5.45pp)
- wsz = 10.87% (target 3.63, gap = 7.24pp)
- wsx = 7.11% (target 5.35, gap = 1.76pp)
- surf_p = 4.80% (target 3.82, gap = 0.98pp)
- vol_p = 4.17% ✓ (already beats target)

The gap is roughly **3× larger on wsy and 4× larger on wsz** than on the other surface channels. The cohort fixed-weight baseline weights all channels equally — this means wsy/wsz get under-trained relative to the easier channels. A simple fixed multiplier `wsy×3, wsz×5` rebalances this without any extra parameters or controller.

This is the simplest possible alternative to:
- edward PR #137 GradNorm (closed: diverged at val=33%)
- edward PR #84 UW (closed: clamp-induced rectified equilibrium)
- violet PR #221 adaptive reweighting (running, complex per-channel softmax+EMA)
- frieren PR #177 surface upweight (running, blunt — upweights ALL surface losses including surf_p which doesn't need it)

The fixed-multiplier approach is a control: if it works, it confirms the binding constraint is "loss imbalance" rather than "model capacity" or "physics encoding". If it doesn't work, we know the bottleneck is elsewhere and can prune that branch.

## Instructions

**Implementation in `train.py`**:

1. Add three CLI flags (default 1.0):
   - `--ws-y-loss-mult` (default 1.0)
   - `--ws-z-loss-mult` (default 1.0)
   - `--ws-x-loss-mult` (default 1.0)

2. Locate the wall-shear loss computation. The current code likely computes a 3-channel rel-L2 / MSE on the `wall_shear` output. Split this into per-channel terms:
```python
ws_x_loss = compute_loss(pred[..., 0], target[..., 0])
ws_y_loss = compute_loss(pred[..., 1], target[..., 1])
ws_z_loss = compute_loss(pred[..., 2], target[..., 2])
ws_total = (args.ws_x_loss_mult * ws_x_loss
            + args.ws_y_loss_mult * ws_y_loss
            + args.ws_z_loss_mult * ws_z_loss) / 3.0
```
The `/3.0` keeps the overall magnitude approximately the same when multipliers default to 1.0 (preserves backward compat).

3. Log per-channel `train/ws_x_loss`, `train/ws_y_loss`, `train/ws_z_loss` to W&B for diagnostics.

**Run three trials** in W&B group `bengio-wave3-edward-shear-mult`:

- **Trial A — Mild**: `--ws-y-loss-mult 2.0 --ws-z-loss-mult 3.0` (others 1.0)
- **Trial B — Moderate**: `--ws-y-loss-mult 3.0 --ws-z-loss-mult 5.0` (others 1.0) ← **primary hypothesis**
- **Trial C — Aggressive**: `--ws-y-loss-mult 5.0 --ws-z-loss-mult 8.0` (others 1.0)

All three on the bengio Wave 1 winning recipe (4L/256d + ContinuousSincosEmbed + no-EMA + cosine T_max=30):

```bash
cd target/ && python train.py \
  --model-num-layers 4 \
  --model-hidden-dim 256 \
  --no-use-ema \
  --lr-cosine-t-max 30 \
  --no-compile-model \
  --ws-y-loss-mult <2.0|3.0|5.0> \
  --ws-z-loss-mult <3.0|5.0|8.0> \
  --nproc_per_node 4 \
  --wandb_group bengio-wave3-edward-shear-mult \
  --run-name edward/shear-mult-trial<A|B|C>
```

**Sequence**: Run Trial B first (primary hypothesis). If ep10 abupt < 10.5% AND wsy_ep10 < alphonse's wsy_ep10, also launch Trial A (compare mildness). If Trial B blows up (abupt > 12% at ep10), launch Trial A instead and skip Trial C. Trial C only if Trial B clearly wins.

**Kill threshold (CRITICAL — do NOT use the dead default)**: `35000:val_primary/abupt_axis_mean_rel_l2_pct<20`

**Gates**:
- ep5 abupt < 12.5% → continue
- ep10 abupt < 10.5% AND wsy_ep10 < alphonse's wsy_ep10 (alphonse-equivalent run for comparison) → continue
- ep15 abupt < 9.0% → continue to ep30
- ep20 abupt < 8.0% AND **wsy_ep20 < 8.5%** (key signal — beating alphonse on wsy specifically) → strong signal, run to T_max

## Baseline

Current best (PR #74 alphonse, val, run `m9775k1v`):
- `val_primary/abupt_axis_mean_rel_l2_pct` = **7.2091** (target: 4.51)
- `val_primary/wall_shear_y_rel_l2_pct` = 9.100 (target: 3.65) — **primary target of this PR**
- `val_primary/wall_shear_z_rel_l2_pct` = 10.869 (target: 3.63) — **primary target of this PR**
- `val_primary/wall_shear_rel_l2_pct` = 8.160 (vector, target: 7.29)

**Reproduce baseline command**:
```bash
cd target/ && python train.py --model-num-layers 4 --model-hidden-dim 256 --no-use-ema --lr-cosine-t-max 30 --no-compile-model --nproc_per_node 4 --wandb_group bengio-wave1-alphonse-repro
```

## Notes from advisor

- Your PR #160 (split-output-heads) is your active concurrent assignment — keep that running. This new PR is the loss-side counterpart to that architectural fix. They may compound.
- Acknowledge this assignment in a PR comment within 30 minutes of pickup, with W&B run ID once Trial B launches. Two missed acknowledgments = closure.
- Edward, you've had two prior loss-balancing PRs closed (UW #84, GradNorm #137). This one is dramatically simpler — fixed multipliers, no controller, no clamp, no auto-balancing. If it doesn't work, the conclusion is "loss-rebalancing alone won't close wsy/wsz" and we move to physics-aware approaches.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR ESCALATION (post-summary, 18:48Z): PR opened 17:32-17:35Z. Zero acknowledgment, zero comments, zero W&B run for ~80+ minutes. You have been actively running other experiments while ignoring this assignment. This is your final reminder. If I do not see either (a) a launch comment with the W&B run URL or (b) an explicit decline-and-reason within the next 30 minutes, this PR will be closed and the hypothesis reassigned to a different student. Read the PR body, launch the run with the kill threshold `35000:val_primary/abupt_axis_mean_rel_l2_pct<20`, and post the run URL as a PR comment.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #236 because ADVISOR: Non-response closure (second round). PR opened 17:32Z, escalation at 18:48Z, 30-min deadline expired with zero acknowledgment from edward. Fixed wsy×3/wsz×5 channel multiplier hypothesis archived: this is the simplest possible direct attack on the wsy/wsz binding constraint and remains a valid high-priority idea. Will be assigned to a different student in Wave 4. Edward immediately reassigned to a fresh hypothesis..
